File: app/src/main/java/zhan/rnadomgirdanimation/refresh/RefreshCallBack.java
package zhan.rnadomgirdanimation.refresh;

import androidx.recyclerview.widget.ListUpdateCallback;
import androidx.recyclerview.widget.RecyclerView;

import java.lang.ref.WeakReference;

public class RefreshCallBack implements ListUpdateCallback {

    private WeakReference<RecyclerView.Adapter> mWeakRefAdapter;

    public RefreshCallBack(RecyclerView.Adapter adapter) {
        mWeakRefAdapter = new WeakReference<>(adapter);
    }

    @Override
    public void onInserted(int position, int count) {
        RecyclerView.Adapter adapter = mWeakRefAdapter.get();
        if (adapter != null) {
            adapter.notifyItemRangeInserted(position, count);
        }
    }

    @Override
    public void onRemoved(int position, int count) {
        RecyclerView.Adapter adapter = mWeakRefAdapter.get();
        if (adapter != null) {
            adapter.notifyItemRangeRemoved(position, count);
        }
    }

    @Override
    public void onMoved(int fromPosition, int toPosition) {
        RecyclerView.Adapter adapter = mWeakRefAdapter.get();
        if (adapter != null) {
            adapter.notifyItemMoved(fromPosition, toPosition);
        }
    }

    @Override
    public void onChanged(int position, int count, Object payload) {
        RecyclerView.Adapter adapter = mWeakRefAdapter.get();
        if (adapter != null) {
            adapter.notifyItemRangeChanged(position, count, payload);
        }
    }
}

File: app/src/main/java/zhan/rnadomgirdanimation/refresh/RefreshCallBack.kt
package zhan.rnadomgirdanimation.refresh

import androidx.recyclerview.widget.ListUpdateCallback
import androidx.recyclerview.widget.RecyclerView

import java.lang.ref.WeakReference

class RefreshCallBack(adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>) : ListUpdateCallback {

    private val mWeakRefAdapter: WeakReference<RecyclerView.Adapter<RecyclerView.ViewHolder>> = WeakReference(adapter)

    override fun onInserted(position: Int, count: Int) {
        val adapter = mWeakRefAdapter.get()
        adapter?.notifyItemRangeInserted(position, count)
    }

    override fun onRemoved(position: Int, count: Int) {
        val adapter = mWeakRefAdapter.get()
        adapter?.notifyItemRangeRemoved(position, count)
    }

    override fun onMoved(fromPosition: Int, toPosition: Int) {
        val adapter = mWeakRefAdapter.get()
        adapter?.notifyItemMoved(fromPosition, toPosition)
    }

    override fun onChanged(position: Int, count: Int, payload: Any?) {
        val adapter = mWeakRefAdapter.get()
        adapter?.notifyItemRangeChanged(position, count, payload)
    }
}
